model/load_balancer.rb

```diff
@@ -53,4 +55,14 @@ def remove_vm(vm)
   def hostname
     "#{name}.#{private_subnet.ubid[-5...]}.#{Config.load_balancer_service_hostname}"
   end
+
+  def need_certificates?
+    return true if certs_dataset.empty?
+
+    certs.all? { |lbc| lbc.created_at && lbc.created_at < Time.now - 60 * 60 * 24 * 30 * 2 }
+  end
+
+  def active_cert
+    certs.sort_by(&:created_at).reverse.find { |lbc| lbc.created_at && lbc.created_at > Time.now - 60 * 60 * 24 * 30 * 3 }
```

Comment:
`created_at` column can't be null. So we don't need check `lbc.created_at` check

```ruby
column :created_at, :timestamp, null: false, default: Sequel.function(:now)
```

Reply:
This also can be done as a single SQL expression: `certs_dataset.where { created_at > Time.now - 60 * 60 * 24 * 30 * 3 }.order(Sequel.desc(:created_at)).first`

But, you may not want to do that, depending how the method is used and whether you want it to be consistent with an already-loaded `certs` array.

Reply:
updated